server/leaderboard/calculate.js

```diff
@@ -1,70 +1,139 @@
 const { workerData, parentPort } = require('worker_threads')
 const utilScores = require('../util/scores')
+const { calcSamples, getPreviousSample } = require('./samples')
 
-const { solves, users, allChallenges } = workerData
+const {
```

Comment:
This is a lot of destructuring, it might be cleaner to refer to the variables as `workerData.XXX`.

Reply:
The variables destructured from `workerData` are used a lot. It's probably less messy to just destructure in in the beginning.